content/blog/2018/06/2018-05-14-simple-pull-request-plugin.adoc

```diff
@@ -125,42 +133,136 @@ using respective paltfroms for their repositories.
 
 === Jenkinsfile.yaml example
 
 Write now no format has been decided for the yaml file. But for this demo I am using below Jenkinsfile.yaml
+[source, yaml]
+----
+agent:
+    dockerImage: maven:3.5.3-jdk-8
+    args: -v /tmp:/tmp
+
+testResultPaths:
+    - target/surefire-reports/*.xml
+
+findBugs: target/*.xml
+
+stages:
+    - name: First
+      scripts:
+        -   ./scripts/hello
+    - name: Build
+      scripts:
+        -   ./scripts/build
+    - name: Tests
+      scripts:
+        -   ./scripts/test
+
+archiveArtifacts:
+    - Jenkinsfile.yaml
+    - scripts/hello.sh
+----
+
+==== Below pipeline code will be generated using above yaml file.
+[source, groovy]
+----
+pipeline {
+  agent {
+    docker {
+      image 'maven:3.5.3-jdk-8'
+      args '-v /tmp:/tmp'
+      alwaysPull false
+      reuseNode false
+    }
+  }
+  stages {
+    stage('First') {
+      steps {
+        script {
+          if (isUnix()) {
+            sh './scripts/hello.sh'
+          } else {
+            bat './scripts/hello.bat'
+          }
+        }
+      }
+    }
+    stage('Build') {
+      steps {
+        script {
+          if (isUnix()) {
+            sh './scripts/build.sh'
+          } else {
+            bat './scripts/build.bat'
+          }
+        }pipeline
+      }
+      post {
+        success {
+          archiveArtifacts artifacts: '**/target/*.jar'
+          archiveArtifacts artifacts: 'Jenkinsfile.yaml'
+          archiveArtifacts artifacts: 'scripts/hello.sh'
+        }
+      }
+    }
+    stage('Tests') {
+      steps {
+        script {
+          if (isUnix()) {
+            sh './scripts/test.sh'
+          } else {
+            bat './scripts/test.bat'
+          }
+        }
+      }
+      post {
+        success {
+          junit 'target/surefire-reports/*.xml'
+        }
+        always {
+          findbugs pattern: 'target/*.xml'
+        }
+      }
+    }
+  }
+}
+----
+==== Pipeline view in Jenkins instance
+
+image:/images/post-images/gsoc-simple-pull-request-plugin/pipeline-view.png[title="Jenkinsfile.yaml", role="center"]
 
-image:/images/post-images/gsoc-simple-pull-request-plugin/YAML.png[title="Jenkinsfile.yaml", role="center"]
-
-Below pipeline code will be generated using above yaml file.
-
-image:/images/post-images/gsoc-simple-pull-request-plugin/Pipeline.png[title="Declarative Pipeline", role="center"]
 
 === Coding Phase 2 plans
 
 . Decide a proper YAML format to use for Jenkinsfile.yaml
-. Create Step Configurator for SPRP plugin. https://issues.jenkins-ci.org/browse/JENKINS-51637[JENKINS-51637].
+. Create Step Configurator for SPRP plugin. https://issues.Jenkins-ci.org/browse/Jenkins-51637[Jenkins-51637].
 This will enable users to use Pipeline steps in Jenkinsfile.yaml.
-. Geting rid of manual tab generation in Pipeline SnippetGenerator class.
+. Getting rid of manual tab generation in Pipeline SnippetGenerator class.
```

Comment:
Automatic indentation generation in the generated Pipeline `SnipperGenerator` class.